test/spmd/test_xla_sharding.py

```diff
@@ -223,9 +223,14 @@ def test_mark_sharding_4d(self):
 
   def test_mark_sharding_partial(self):
     device = xm.xla_device()
-    t1 = torch.randn(4, 4).to(xm.xla_device())
-    t2 = torch.randn(4, 4).to(xm.xla_device())
-    expected = (t1 @ t2).cpu()
+    t1 = torch.randn(4, 4).to(device)
+    t2 = torch.randn(4, 4).to(device)
+    # Somehow the eager cpu result is different from the xla result.
```

Comment:
hmm... I expect them to be off by a bit, due to optimization. They should not be too different through.

Reply:
Hmm... Actually quite off. Probably worth looking next.